### PR TITLE
Password manager not implemented

### PR DESCRIPTION
**Pull request by Gavinok**
**Feature description**
We need to implement a password manager to accommodate for password changes on admin/staff/user accounts. This should be added to the account path.

**Proposed solution**
1. Add /updatepassword backend endpoint to logincontroller
2. Add user.get() to userInfoDAO to retrieve current user info
3. Change the user password of the retrieved user
4. Send signout back to front end to force user to re-sign in.

**Describe alternatives you've considered**
* Might want to add minimum password requirements

**Additional context**
Stack overflow on how to accomplish this: https://stackoverflow.com/questions/11881479/how-do-i-update-an-entity-using-spring-data-jpa

**Story points**
13

### CODE REVIEW

File: src/test/java/com/uvic/venus/AdminTests.java [hunk 1]
```diff
@@ -67,7 +67,8 @@ void fetchAllUsersSingleUser() throws Exception {
 
         // Mock for loading the user for lookup
         List<UserInfo> userlist = new ArrayList<UserInfo>();
-        userlist.add(new UserInfo(userInfo.getUsername(), userInfo.getFirstName(), userInfo.getLastName()));
+        List<UserInfo> userlist2 = userlist;
```
Comment (bmazerolle) [inline src/test/java/com/uvic/venus/AdminTests.java:70]: Not following why we need a second object here, why not just append to `userlist`?

Comment (bmazerolle) [inline src/test/java/com/uvic/venus/AdminTests.java:70]: Not following why we need a second object here, why not just append to userlist?

File: src/test/java/com/uvic/venus/PasswordTests.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    void updatePasswordUnauthorized() throws Exception {
+        passController = spy(new PasswordController());
+        MockitoAnnotations.openMocks(this);
+
+        assertThat(passController.updatePassword("hello").getBody())
+            .isEqualTo("User Not Found");
+    }
+
+    @Test
+    void updatePasswordAuthorized() throws Exception {
```
Comment (bmazerolle) [inline src/test/java/com/uvic/venus/PasswordTests.java:67]: Easy enough to add an unauthorized one too?

File: src/main/java/com/uvic/venus/controller/PasswordController.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+        if(auth == null)
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("User Not Found");
+
+	UserDetails userDetails = loadUserByUsername(manager, auth.getName());
+	System.out.println(userDetails.getAuthorities());
+
+	User.UserBuilder builder = User.builder();
+        builder.username(userDetails.getUsername());
+        // builder.password(userDetails.getPassword());
```
Comment (bmazerolle) [inline src/main/java/com/uvic/venus/controller/PasswordController.java:44]: Remove

File: src/main/java/com/uvic/venus/controller/PasswordController.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    @PostMapping(value = "/updatepassword")
+    public ResponseEntity<?> updatePassword(String password){
+        Authentication auth = SecurityContextHolder.getContext().getAuthentication();
+	JdbcUserDetailsManager manager = new JdbcUserDetailsManager(dataSource);
+
+        if(auth == null)
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body("User Not Found");
+
+	UserDetails userDetails = loadUserByUsername(manager, auth.getName());
+	System.out.println(userDetails.getAuthorities());
```
Comment (bmazerolle) [inline src/main/java/com/uvic/venus/controller/PasswordController.java:40]: Not sure if we should keep println()s in

**Review by bmazerolle — CHANGES_REQUESTED** [review]

File: src/test/java/com/uvic/venus/AdminTests.java (hunk 1 above)
Comment (Gavinok) [reply src/test/java/com/uvic/venus/AdminTests.java:70]: Definitely the lack of sleep for most of these.

**Review by bmazerolle — APPROVED** [review]

**Pull request merged**